File: frontend/src/types/report.ts
export interface DimensionBreakdown {
  dimension: string;
  score: number;
  weight: number;
  strengths: string[];
  weaknesses: string[];
}

export interface StrengthItem {
  text: string;
  quote: string;
}

export interface WeaknessItem {
  text: string;
  quote: string;
}

export interface ImprovementSuggestion {
  dimension: string;
  priority: "high" | "medium" | "low";
  suggestion: string;
  example?: string;
}

export interface SessionReport {
  session_id: string;
  scenario_name: string;
  product: string;
  hcp_name: string;
  overall_score: number;
  passed: boolean;
  feedback_summary: string;
  duration_seconds: number | null;
  completed_at: string | null;
  dimensions: DimensionBreakdown[];
  strengths: StrengthItem[];
  weaknesses: WeaknessItem[];
  improvements: ImprovementSuggestion[];
  key_messages_delivered: number;
  key_messages_total: number;
}

export interface SuggestionResponse {
  id: string;
  dimension: string;
  suggestion: string;
  priority: "high" | "medium" | "low";
}

export interface ScoreHistoryItem {
  session_id: string;
  scenario_name: string;
  overall_score: number;
  passed: boolean;
  completed_at: string;
  dimensions: Array<{
    dimension: string;
    score: number;
    weight: number;
    improvement_pct: number | null;
  }>;
}

File: backend/app/schemas/report.py
"""Post-session report schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class StrengthItem(BaseModel):
    """A strength identified in the session."""

    text: str
    quote: str | None = None

    model_config = ConfigDict(from_attributes=True)


class WeaknessItem(BaseModel):
    """A weakness identified in the session."""

    text: str
    quote: str | None = None

    model_config = ConfigDict(from_attributes=True)


class ImprovementSuggestion(BaseModel):
    """Actionable improvement suggestion."""

    dimension: str
    suggestion: str
    priority: str = "medium"  # high / medium / low

    model_config = ConfigDict(from_attributes=True)


class DimensionBreakdown(BaseModel):
    """Per-dimension score breakdown for the report."""

    dimension: str
    score: float
    weight: int
    max_score: float = 100.0
    strengths: list[StrengthItem]
    weaknesses: list[WeaknessItem]
    suggestions: list[str]

    model_config = ConfigDict(from_attributes=True)


class SessionReport(BaseModel):
    """Complete post-session report."""

    session_id: str
    scenario_name: str
    product: str
    hcp_name: str
    overall_score: float
    passed: bool
    feedback_summary: str
    duration_seconds: int | None = None
    completed_at: datetime | None = None
    dimensions: list[DimensionBreakdown]
    strengths: list[StrengthItem]
    weaknesses: list[WeaknessItem]
    improvements: list[ImprovementSuggestion]
    key_messages_delivered: int = 0
    key_messages_total: int = 0

    model_config = ConfigDict(from_attributes=True)
